doc/crypto/api/keys/management.rst

```diff
@@ -412,6 +413,114 @@ When creating a key, the attributes for the new key are specified in a `psa_key_
+    .. param:: size_t label_length
+        Size of the ``label`` buffer in bytes.
+    .. param:: psa_key_id_t * key
+        On success, an identifier for the newly created key.
+        `PSA_KEY_ID_NULL` on failure.
+
+    .. return:: psa_status_t
+    .. retval:: PSA_SUCCESS
+        Success.
+    .. retval:: PSA_ERROR_DOES_NOT_EXIST
+        ``label`` does not refer to key material within the location specified in ``attributes``.
+    .. retval:: PSA_ERROR_NOT_SUPPORTED
+        The key attributes, as a whole, are not supported, either by the implementation in general or in the specified storage location.
+    .. retval:: PSA_ERROR_INVALID_ARGUMENT
+        The following conditions can result in this error:
+
+        *   The key type is invalid.
+        *   The key size is nonzero, and is incompatible with the stored key material.
+        *   The key lifetime specifies a non-volatile persistence level.
+        *   The key lifetime specifies an invalid storage location.
+        *   The key identifier in the provided attributes is not `PSA_KEY_ID_NULL`.
+        *   The key usage flags include invalid values.
+        *   The key's permitted-usage algorithm is invalid.
+        *   The key attributes, as a whole, are invalid.
+        *   The stored key material is incompatible with the provided key attributes.
+    .. retval:: PSA_ERROR_NOT_PERMITTED
+        The implementation does not permit creating a key with the specified attributes due to some implementation-specific policy.
+    .. retval:: PSA_ERROR_INSUFFICIENT_MEMORY
+    .. retval:: PSA_ERROR_COMMUNICATION_FAILURE
+    .. retval:: PSA_ERROR_STORAGE_FAILURE
+    .. retval:: PSA_ERROR_DATA_CORRUPT
+    .. retval:: PSA_ERROR_DATA_INVALID
+    .. retval:: PSA_ERROR_CORRUPTION_DETECTED
+    .. retval:: PSA_ERROR_BAD_STATE
+        The library requires initializing by a call to `psa_crypto_init()`.
+
+    This function allows applications to register existing key material that has been provisioned outside the |API|, such as during manufacturing or by a secure element.
+    It can also be used to create a key identifier for keys that are deterministically derived from a secret within the implementation.
+    After registering the key, the application has a volatile key identifier that can be used in cryptographic operations permitted by its usage flags and algorithm policy.
+
+    The key material is identified by its location, specified in the provided attributes ``lifetime`` value, and the ``label`` parameter.
+    The format of the label is specific to the implementation and storage location.
+    Typically, the label is used as a location-specific identifier for the key material, or can provide input for deriving the key material from an internal secret.
+
+    This function can only be used to create a volatile key.
```

Comment:
Somewhere (maybe not here) we should state that the behavior with persistent lifetimes is reserved and may be specified in a future version of the specification, to avoid divergent implementation-specific extensions.

Reply:
I think perhaps an Implementation Note here might work